base/parse.jl

```diff
@@ -126,27 +130,49 @@ end
 
 function tryparse_internal(::Type{Bool}, sbuff::Union{String,SubString},
         startpos::Int, endpos::Int, base::Integer, raise::Bool)
-    len = endpos-startpos+1
-    p = pointer(sbuff)+startpos-1
+    null = Nullable{Bool}()
+
+    if isempty(sbuff)
+        raise && throw(ArgumentError("input string is empty"))
+        return null
+    end
+
+    len = endpos - startpos + 1
+    p   = pointer(sbuff) + startpos - 1
     (len == 4) && (0 == ccall(:memcmp, Int32, (Ptr{UInt8}, Ptr{UInt8}, UInt),
         p, "true", 4)) && (return Nullable(true))
     (len == 5) && (0 == ccall(:memcmp, Int32, (Ptr{UInt8}, Ptr{UInt8}, UInt),
         p, "false", 5)) && (return Nullable(false))
-    raise && throw(ArgumentError("invalid Bool representation: " *
-        repr(SubString(sbuff, startpos, endpos))))
-    Nullable{Bool}()
+
+    substr = SubString(sbuff, startpos, endpos)
+    if count(isspace, substr) == len # all chars were whitespace
+        raise && throw(ArgumentError("input string only contains whitespace"))
```

Comment:
This looks a bit weird stylistically. Why not wrap both branches inside an `if raise` block?